src/components/app.cjsx

```diff
@@ -1,35 +1,140 @@
 React = require 'react'
 BS = require 'react-bootstrap'
+classnames = require 'classnames'
 Exercise = require './exercise'
+ErrorModal = require './error-modal'
+{ExerciseActions, ExerciseStore} = require '../stores/exercise'
+AsyncButton = require 'openstax-react-components/src/components/buttons/async-button.cjsx'
 
 module.exports = React.createClass
   displayName: 'App'
-  getInitialState: -> {exerciseId: 'new'}
+  getInitialState: -> {verticalDisplay: true}
 
-  update: (event) ->
-    exerciseId = event.target.value
+  update: -> @forceUpdate()
+
+  componentWillMount: ->
+    ExerciseStore.addChangeListener(@update)
+
+  componentWillUnmount: ->
+    ExerciseStore.removeChangeListener(@update)
+
+  componentDidMount: ->
+    if @props.exerciseId is 'new'
+      @addNew()
+    else if @props.exerciseId
+      this.refs.exerciseId.getDOMNode().value = @props.exerciseId
```

Comment:
great catch! I think it's probably fairly dangerous to rely on refs being set in a `didMount`